scaleway/helpers_rdb_test.go

```diff
@@ -0,0 +1,115 @@
+				IP: net.IPv4(0x1, 0x1, 0x1, 0x1), Mask: net.IPMask{0xff, 0xff, 0xff, 0xff}}}, Zone: scw.ZoneFrPar1},
+			B: &rdb.EndpointPrivateNetworkDetails{PrivateNetworkID: "6ba7b810-9dad-11d1-80b4-00c04fd430c8", ServiceIP: scw.IPNet{IPNet: net.IPNet{
+				IP: net.IPv4(0x1, 0x1, 0x1, 0x1), Mask: net.IPMask{0xff, 0xff, 0xff, 0xff}}}, Zone: scw.ZoneFrPar1},
+			expected: true,
+		},
+		{
+			name: "notEqualIP",
+			A: &rdb.EndpointPrivateNetworkDetails{PrivateNetworkID: "6ba7b810-9dad-11d1-80b4-00c04fd430c8", ServiceIP: scw.IPNet{IPNet: net.IPNet{
+				IP: net.IPv4(0x1, 0x1, 0x1, 0x1), Mask: net.IPMask{0xff, 0xff, 0xff, 0xff}}}, Zone: scw.ZoneFrPar1},
+			B: &rdb.EndpointPrivateNetworkDetails{PrivateNetworkID: "6ba7b810-9dad-11d1-80b4-00c04fd430c8", ServiceIP: scw.IPNet{IPNet: net.IPNet{
+				IP: net.IPv4(0x1, 0x1, 0x1, 0x2), Mask: net.IPMask{0xff, 0xff, 0xff, 0xff}}}, Zone: scw.ZoneFrPar1},
+			expected: false,
+		},
+		{
+			name: "notEqualZone",
+			A: &rdb.EndpointPrivateNetworkDetails{PrivateNetworkID: "6ba7b810-9dad-11d1-80b4-00c04fd430c8", ServiceIP: scw.IPNet{IPNet: net.IPNet{
+				IP: net.IPv4(0x1, 0x1, 0x1, 0x1), Mask: net.IPMask{0xff, 0xff, 0xff, 0xff}}}, Zone: scw.ZoneFrPar1},
+			B: &rdb.EndpointPrivateNetworkDetails{PrivateNetworkID: "6ba7b810-9dad-11d1-80b4-00c04fd430c8", ServiceIP: scw.IPNet{IPNet: net.IPNet{
+				IP: net.IPv4(0x1, 0x1, 0x1, 0x1), Mask: net.IPMask{0xff, 0xff, 0x0, 0x0}}}, Zone: scw.ZoneFrPar2},
+			expected: false,
+		},
+		{
+			name: "notEqualMask",
+			A: &rdb.EndpointPrivateNetworkDetails{PrivateNetworkID: "6ba7b810-9dad-11d1-80b4-00c04fd430c8", ServiceIP: scw.IPNet{IPNet: net.IPNet{
+				IP: net.IPv4(0x1, 0x1, 0x1, 0x1), Mask: net.IPMask{0xff, 0xff, 0xff, 0xff}}}, Zone: scw.ZoneFrPar1},
+			B: &rdb.EndpointPrivateNetworkDetails{PrivateNetworkID: "6ba7b810-9dad-11d1-80b4-00c04fd430c8", ServiceIP: scw.IPNet{IPNet: net.IPNet{
+				IP: net.IPv4(0x1, 0x1, 0x1, 0x1), Mask: net.CIDRMask(24, 32)}}, Zone: scw.ZoneFrPar1},
+			expected: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equal(t, tt.expected, isEndPointEqual(tt.A, tt.B))
+		})
+	}
+}
+
+func TestGetRdbEndpointsToRemove(t *testing.T) {
+	tests := []struct {
+		name      string
+		Endpoints []*rdb.Endpoint
+		Updates   []interface{}
+		Expected  map[string]bool
+	}{
+		{
+			name: "removeAll",
+			Endpoints: []*rdb.Endpoint{&rdb.Endpoint{
+				ID: "6ba7b810-9dad-11d1-80b4-00c04fd430c1",
+				PrivateNetwork: &rdb.EndpointPrivateNetworkDetails{PrivateNetworkID: "6ba7b810-9dad-11d1-80b4-00c04fd430c8",
+					ServiceIP: scw.IPNet{IPNet: net.IPNet{
+						IP: net.IPv4(0x1, 0x1, 0x1, 0x1), Mask: net.IPMask{0xff, 0xff, 0xff, 0xff}}},
+					Zone: scw.ZoneFrPar1},
+			}},
+			Expected: map[string]bool{
+				"6ba7b810-9dad-11d1-80b4-00c04fd430c1": true,
+			},
+		},
+		{
+			name: "shouldUpdatePrivateNetwork",
```

Comment:
Could you add a test where we check if we should add a private network at a place where none existed before?